objects/screen.c

```diff
@@ -94,12 +125,41 @@
  * @signal swapped
  */
 
+/** This signal is emitted when the list of physical screen viewport changes.
+ *
+ * Each viewport in the list corresponds to a **physical** screen rectangle, which
+ * is **not** the `viewports` property of the `screen` objects.
+ *
+ * Each entry in the `viewports` entry has the following keys:
+ *
+ * * `geometry` *(table)*: A table with an `x`, `y`, `width` and `height` keys.
+ * * `outputs` *(table)*: All outputs sharing this viewport.
+ * * `maximum_dpi` *(number)*: The DPI of the most dense output.
+ * * `minimum_dpi` *(number)*: The DPI of the least dense output.
+ * * `preferred_dpi` *(number)*: The optimal DPI.
+ *
+ * @signal property::viewports
+ * @tparam table viewports A table containing all physical viewports.
+ * @see automatic_factory
+ */
```

Comment:
There is no such signal. `property::viewports` only appears four times in comments. Could you please re-read the documentation that you are adding and see if it still applies? (Or tell me where this `property::viewports` is emitted?)

Reply:
It is emitted from Lua. It's a proxy for `property::_viewports`. The difference is that `property::viewports` is sent after `awful` (dpi.lua) had a chance to modify the values (which, in this case, means adding the DPI). It was also added for the tests since it was easier to monkey-patch awful to set the test values. The doc is in the original commit to avoid having to add it in one commit and move it to `awful` in the next commit. That would make the diff noisier.